### ISSUE TITLE
Refactor models

### ISSUE THREAD

**Issue by alessandropellegrini** [HPDCS/ROOT-Sim#16]
Models should be reorganized, so as to be more customizable. This will help a lot in devising experiments for the papers, and for (future) test cases.

This issue depends on #15 as we definitely need macros to simplify rewriting everything.

**Comment by alessandropellegrini** [comment]
Airport has several flaws. The first ones that I've spot are:
- ScheduleNewEvent() is called in OnGVT() [partially fixed in commit 732ff1892cc53ec44430cce6979efec743703dc0]
- There are support functions which pass as the payload of a message a pointer to the internal state
- There are a lot of pointless malloc() calls in the code
- Some malloc() calls for temporary buffers don't have the corresponding free()
- Routes as LPs are not really useful as they are now

**Comment by alessandropellegrini** [comment]
Remember to introduce `Makefile`s to simplify the model build process.

**Comment by alessandropellegrini** [comment]
`Makefile`s for models have been introduced in commit 8a8a88529cdabde423901162c7f7b25d56c38fbe.

**Comment by alessandropellegrini** [comment]
There are models, such as `collector`, which are in the `/models` just to support specific coverage tests. These should be moved to `/tests`, to leave in `/models` only the real models.

**Comment by alessandropellegrini** [comment]
On hold, planned for v3.x
